indexer-app/src/cmr/indexer/services/index_service.clj

```diff
@@ -176,18 +177,52 @@
 
 (def REINDEX_BATCH_SIZE 2000)
 
+(defn- positions
+  "Return a map of the index of each item that matches the predicate"
+  [pred coll]
+  (keep-indexed (fn [idx x]
+                  (when (pred x)
+                    idx))
+                coll))
+
+(defn- science-keywords->elastic-docs
+  "Convert hierarchical science-keywords to colon-separated elastic docs for indexing"
+  [science-keywords]
+  (let [keyword-hierarchy [:topic :term :variable-level-1
+                           :variable-level-2 :variable-level-3 :detailed-variable]
+        terminal-key (->> keyword-hierarchy
+                          (map #(get science-keywords %))
+                          (positions nil?))
+        keyword-string (->> keyword-hierarchy
+                            (map #(get science-keywords %))
+                            (remove nil?)
+                            (s/join ":"))
+        keyword-value (get science-keywords terminal-key)]
+     {:type "science_keywords"
+      :_id (str (s/lower-case keyword-string) "_science_keywords")
```

Comment:
How big can this string be? `_id` is limited to 512 bytes.

Reply:
It's never going to be anything too crazy - the values are humanized

Reply:
The following value is 85 bytes before I convert it to a hash and is a pretty typical length for a science keyword
`Biosphere:Ecological Dynamics:Ecosystem Functions:Respiration Rate:Evapotranspiration`

Reply:
Okay cool - thanks for implementing it anyway :)